### DISCUSSION TITLE
There are no models in the Сontrolnet

### DISCUSSION THREAD

**Discussion by AntonKrendel** [lllyasviel/stable-diffusion-webui-forge#97] [Q&A]
Hi. I'm faced with the problem that I don't have models displayed in the controlnet. Can I take the model file and put it in some folder so that the model appears? 

![image](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/96669001/a5926193-69e5-4635-87e7-f57830c96b7e)

**Comment by phleshdef** [comment]
I'm surprised and a little disappointed that this question sat here for 2 days without anyone answering it because its a super simple answer (I figured it out myself after a little trial and error). 

Its [root web ui]/models/Controlnet

**Reply by krummrey** [reply]
I've tried 
```ln -s <auto1111home>/extensions/sd-webui-controlnet/models <forgehome>/models/ControlNet/models```

That worked for me.

**Comment by yamfun** [comment]
Is there an arg for sharing IP Adaptor model folder path with a1111?

Edit: Ignore me, simply set the path in settings "Extra path to scan for ControlNet models (e.g. training output directory)"

**Reply by IndigoDosSantos** [reply]
I don't know whether there are args for this, but you can use symlinks.

**Reply by tmprabubiz** [reply]
How to use symlinks to use controlnet models including ip adopter models installed in A1111 webui folder. May you please explain in detail since I am not good in coding. Thank You

**Reply by cibernicola** [reply]
`--controlnet-dir "PATH" `

**Comment by PAEz** [comment]
@tmprabubiz 
> How to use symlinks to use controlnet models including ip adopter models installed in A1111 webui folder. May you please explain in detail since I am not good in coding. Thank You

Symlinks allow you to create a link to another location.  So you can create a link that acts as a directory to another directory.  So you can create a link called Models in any directory and point it to where you store all your models.  Then if any app tries to to access the link as a directory it is redirected to the directory that link points to.

Heres a nice article on it....
https://www.howtogeek.com/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/

**Reply by tmprabubiz** [reply]
Thank You. I understood nothing but just aware of a name 'symlinks' inspite of reading the article in your link. Dear techies, please speak in simple language with non-technical people. Give an example file with the so called symlink. Or just mention how to do that in the webui-user.bat file. 
Please check my webui-user.bat file lines below (in which I could not create a link for the Controlnet files in my A1111)
___________________________________________________________________
@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS= --xformers --always-offload-from-vram --disable-in-browser --always-normal-vram --no-gradio-queue --ckpt-dir "C:\Users\Winnertmp\a1111\stable-diffusion-webui\models\Stable-diffusion" --lora-dir "C:\Users\Winnertmp\a1111\stable-diffusion-webui\models\Lora" --hypernetwork-dir "C:\Users\Winnertmp\a1111\stable-diffusion-webui\models\hypernetworks" --embeddings-dir "C:\Users\Winnertmp\a1111\stable-diffusion-webui\embeddings" --vae-dir "C:\Users\Winnertmp\a1111\stable-diffusion-webui\models\VAE" --gfpgan-model "C:\Users\Winnertmp\a1111\stable-diffusion-webui\models\GFPGAN" --theme dark

start Firefox:http://127.0.0.1:7860/?__theme=dark
call webui.bat
_____________________________________________________________________________

You need not bother, I will wait for someone or the developer to comeup with the solution. Otherwise I will just copy my 57GB Controlnet files from A1111

**Reply by dermesut** [reply]
have you tried --forge-ref-a1111-home C:\Users\Winnertmp\a1111\stable-diffusion-webui instead of those individual flags? (assuming C:\Users\Winnertmp\a1111\stable-diffusion-webui is your "other" a1111-installation outside of forge - and: i believe gfpgan would still need a flag to be set)

**Reply by tmprabubiz** [reply]
To make things easy in Forge I understand the following

In A1111 all controlnet models can be placed in the following folder ''''stable-diffusion-webui\models\ControlNet''''

No need to place the controlnet models in ''''stable-diffusion-webui\extensions\sd-webui-controlnet\models''''

With the above changes and other conversations I made my webui-user.bat as below

_______________________________________________________________________
@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS= --forge-ref-a1111-home "C:\Users\REVIVE\A1111\stable-diffusion-webui"   --gfpgan-model "C:\Users\REVIVE\A1111\stable-diffusion-webui\models\GFPGAN"--always-normal-vram --always-offload-from-vram --xformers --theme dark

@REM Uncomment following code to reference an existing A1111 checkout.
@REM set A1111_HOME=Your A1111 checkout dir
@REM
@REM set VENV_DIR=%A1111_HOME%/venv
@REM set COMMANDLINE_ARGS=%COMMANDLINE_ARGS% ^
@REM  --ckpt-dir %A1111_HOME%/models/Stable-diffusion ^
@REM  --hypernetwork-dir %A1111_HOME%/models/hypernetworks ^
@REM  --embeddings-dir %A1111_HOME%/embeddings ^
@REM  --lora-dir %A1111_HOME%/models/Lora

call webui.bat

___________________________________________
Since the controlnet directory is set to the A1111 main models directory forge easily access that. Since somebody said gfgan needs a separate commandline arguement to make things easier. As of now my problems are over. I have to venture photomaker.

**Comment by rdeforest** [comment]
So what's the final word on ControlNet models in SDWF? Install a1111 and tell SDWF to use the ones in the a1111 directory?
